prime-router/src/main/kotlin/Report.kt

```diff
@@ -487,6 +473,74 @@ class Report {
+                    it.testingLabCity = row.getStringOrNull("testing_lab_city")
+                    it.testingLabClia = row.getStringOrNull("testing_lab_clia")
+                    it.testingLabCounty = row.getStringOrNull("testing_lab_county")
+                    it.testingLabName = row.getStringOrNull("testing_lab_name")
+                    it.testingLabPostalCode = row.getStringOrNull("testing_lab_zip_code")
+                    it.testingLabState = row.getStringOrNull("testing_lab_state")
+                    it.patientCounty = row.getStringOrNull("patient_county")
+                    it.patientEthnicityCode = row.getStringOrNull("patient_ethnicity")
+                    it.patientEthnicity = metadata.findValueSet("hl70189")
+                        ?.toDisplayFromCode(it.patientEthnicityCode)
+                    it.patientGenderCode = row.getStringOrNull("patient_gender")
+                    it.patientGender = metadata.findValueSet("hl70001")?.toDisplayFromCode(it.patientGenderCode)
+                    it.patientPostalCode = row.getStringOrNull("patient_zip_code")
+                    it.patientRaceCode = row.getStringOrNull("patient_race")
+                    it.patientRace = metadata.findValueSet("hl70005")?.toDisplayFromCode(it.patientRaceCode)
+                    it.patientState = row.getStringOrNull("patient_state")
+                    it.testResultCode = row.getStringOrNull("test_result")
+                    it.testResult = metadata.findValueSet("covid-19/test_result")
+                        ?.toDisplayFromCode(it.testResultCode)
+                    it.equipmentModel = row.getStringOrNull("equipment_model_name")
+                    it.specimenCollectionDateTime = row.getStringOrNull("specimen_collection_date_time").let { dt ->
+                        if (!dt.isNullOrEmpty()) {
+                            try {
+                                LocalDate.parse(dt, Element.datetimeFormatter)
+                            } catch (_: Exception) {
+                                null
+                            }
+                        } else {
+                            null
+                        }
+                    }
+                    it.patientAge = row.getStringOrNull("patient_dob").let { dob ->
+                        try {
+                            val d = LocalDate.parse(dob, Element.dateFormatter)
+                            if (d != null && it.specimenCollectionDateTime != null) {
+                                Period.between(d, it.specimenCollectionDateTime).years.toString()
+                            } else {
+                                null
+                            }
+                        } catch (_: Exception) {
+                            null
+                        }
+                    }
+                    it.reportId = this.id
+                }
+            }
+        } catch (e: Exception) {
+            logger.warn(e)
+            emptyList()
+        }
+    }
+
```

Comment:
This gets all the data from the report for our insert. I'm not super crazy about how I'm doing the valueset translation, but I'm not sure there's a cleaner way to accomplish it.

Reply:
I wonder, since we are marking the fields in the schema as PII, could we get these fields based on that?  Of course, the different types makes it more difficult.

Reply:
It's not about the types that make it difficult, it's the fact that there's lots of non-PII/non-PHI data we might not want. I am trying to keep this tightly constrained to a smaller subset of data. I recognize there is a tradeoff there that we might need to update this code as we add new column types, but for now, I think that that is less of an issue, and is manageable.